### ISSUE TITLE
[Feature] Need a way to get Tracing either from OpenTelemetry or ActivitySources, I don't see either in the codebase.

### ISSUE THREAD

**Issue by DR9885** [connamara/quickfixn#880]
Need a way to get tracing either from OpenTelemetry or ActivitySources, I don't see either in the codebase.

If I did happen to miss them, just provide the way to enable them.

If there a different way I could do this with middleware, to have trigger just before message is picked up, and after message is handled. I could add my own Tracing.

**Comment by gbirchmeier** [comment]
We don't have it.  I'm not opposed to adding it somehow, but I also don't really know much about it.

**Comment by DR9885** [comment]
Tracing can be done by either ActivitySource or OpenTelemetry. 
- ActivitySource - Part of C# framework, and works for any shipper
- OpenTelemetry - an open source standard, that can do BOTH tracing and shipping.

Note: You wont need these for the framework, but to test the data it makes it easier
Tracing Shippers (might not be the right term, but they collect the trace data and batch it up to be sent)
- OpenTelemetry - A Standard That almost all shippers are moving to (ElasticSearch, DataDog, Sentry, etc...)

To summarize though, I'd recommend using ActivitySource to Write the traces, and then just test that they work with OpenTelemetry Console Logging. Which can be helpful.

Links: (These Pages will be very helpful)
https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing-instrumentation-walkthroughs
https://opentelemetry.io/docs/languages/net/getting-started/

**Comment by gbirchmeier** [comment]
If someone wants to collaborate, please reach out.  But I'm not going to write this for free all on my own (unless I happen to need it myself)
